packages/components/src/organisms/SKUMatrix/provider/SKUMatrixProvider.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { createContext, useCallback, useState } from 'react'
+import type { ReactNode } from 'react'
+
+interface IAllVariantProducts {
+  id: string
+  name: string
+  image: {
+    url: string
+    alternateName: string
+  }
+  inventory: number
+  availability: string
+  price: number
+  listPrice: number
+  priceWithTaxes: number
+  listPriceWithTaxes: number
+  specifications: Record<string, string>
+  selectedCount: number
+  offers: {
+    highPrice: number
+    lowPrice: number
+    lowPriceWithTaxes: number
+    offerCount: number
+    priceCurrency: string
+    offers: Array<{
+      listPrice: number
+      listPriceWithTaxes: number
+      sellingPrice: number
+      priceCurrency: string
+      price: number
+      priceWithTaxes: number
+      priceValidUntil: string
+      itemCondition: string
+      availability: string
+      quantity: number
+    }>
+  }
+}
+
+export interface SKUMatrixProviderContextValue {
+  /*
+    value that indicates if the modal is open
+  */
+  isOpen: boolean
+  /*
+    array of all variant products
+  */
+  allVariantProducts: IAllVariantProducts[]
+  /*
+    function to set the array of all variant products
+  */
+  setAllVariantProducts(
+    items: React.SetStateAction<IAllVariantProducts[]>
+  ): void
+  /*
+  */
+  handleChangeQuantityItem(id: string, value: number): IAllVariantProducts[]
```

Comment:
looks like you don't use the [return of this function](https://github.com/cubos-vtex/faststore/blob/ff4c142e22fc1b2af1b64d01bb4469179d349f11/packages/components/src/organisms/SKUMatrix/SKUMatrixSidebar.tsx#L98), only updates the state. What about?
```suggestion
  handleChangeQuantityItem(id: string, value: number): void
```

I also agree with @hellofanny suggestion :)